index.js

```diff
@@ -430,12 +474,15 @@ function pickRelicLocation(ctx, locs) {
 
   // Find a location not locked by this current relic
   locationsAvailable = locationsAvailable.filter(function(loc) {
+    // Restrict certain locations for certain relics
     if (relic.locationBlacklist
         && relic.locationBlacklist.indexOf(loc.location) !== -1) {
       return false
     }
+
+    // Find a location with a lock set that doesn't contain this relic
     return loc.locks.some(function(lock) {
-      return lock.indexOf(relic.ability) === -1
+      return !lock.has(relic.ability)
```

Comment:
`Array.prototype.has` isn't a standard method, guessing you meant `include`. Even then, let's try to avoid ES6 additions where possible to increase browser compatibility. Please revert to the original use of `indexOf`.

Reply:
I use `has` because they've been changed into Sets rather than Arrays